dcicutils/ff_utils.py

```diff
@@ -457,65 +457,150 @@ def faceted_search(key=None, ff_env=None, item_type=None, **kwargs):
     return search_metadata(search, ff_env=ff_env, key=key)
 
 
-def get_associated_qc_metrics(uuid, key=None, ff_env=None,
-                              exclude_raw_files=True,
-                              exclude_supplementary_files=True):
+def fetch_files_qc_metrics(data, associated_files=['processed_files'], ignored_fields=True, key=None, ff_env=None):
     """
-    Given a uuid of an experiment set, return a dictionary of uuid : item
-    mappings of quality metric items.
+    Utility function to grap all the qc metrics from associated types of file such as:
+    'proccessed_files', 'other_processed_files', 'files'
+    inputs:
+        data: the metadata of a ExperimentSet or Experiment
+        associated_files: a list of the types of the files fields the qc metrics will be extracted from:
+            examples are = ['files', 'processed_files', 'other_processed_files']
+    Args:
+        ignored_fields: flag to ignore 4DN custom fields from the qc metric object
+
+    Returns:
+        a dictionary of dictionaries containing the qc_metric information
+    """
+    qc_metrics = {}
+
+    if ignored_fields:
+        ignored_qc_fields = ['contributing_labs', 'schema_version', 'external_references', '@context', 'aliases',
```

Comment:
This list should be separated out into a constant at file toplevel so that it can be inspected by people wondering what's in it. You might also want to make it be a tuple in that case, just so no one is tempted to modify it. But let's assume you've done:
```
IGNORABLE_QC_FIELDS = ('contributing_labs', ...)
```
Then here you might want to do something like this (untested, just offered as a sketch):
```
if isinstance(ignored_fields, (list, tuple, set)):
    ignored_qc_fields = ignored_fields
elif isinstance(ignored_fields, bool):
    ignored_qc_fields = IGNORABLE_QC_FIELDS if ignored_fields else []
else:
    raise ValueError("Expected ignored_fields to be True, False, or a collection of field names: %s"
                     % ignored_fields)
# If we're going to a lot of membership tests, it will be faster as a set
ignored_fields.= set(ignored_fields).
```